src/modules/data_module.c

```diff
@@ -621,43 +623,56 @@ static void on_cloud_state_connected(struct data_msg_data *msg)
 			config_change = true;
 		}
 
-		if (current_cfg.actw != new.actw && new.actw != 0) {
+		if (current_cfg.gpst != new.gpst &&
+		    new.gpst > 0) {
+			current_cfg.gpst = new.gpst;
+			LOG_WRN("New GPS timeout: %d", current_cfg.gpst);
+
+			config_change = true;
+		}
+
+		/* Only apply a new active wait timeout configuration if the
+		 * new active wait timeout exceeds the current GPS timeout value
+		 * + 10 seconds. This is to hinder requesting GPS data from the
+		 * GPS module in the middle of a GPS search.
+		 */
+		if (current_cfg.actw != new.actw &&
+		    new.actw > current_cfg.gpst + 10 &&
+		    new.actw > 0) {
```

Comment:
Perhaps print warnings if incoming configs are out of bounds?

Reply:
Agreed.